### PR TITLE
Fix FDSNWS missing data

### PR DESCRIPTION
**Pull request by Brtle**
This PR fixes the issue discussed [here](https://forum.seiscomp3.org/t/fdsnws-missing-data/1260).

The `start_time` of the miniseed files wasn't handled properly in the [isEnd](https://github.com/SeisComP3/seiscomp3/blob/master/src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp#L410) method. These changes improve things but do not take into account microseconds, only 1e-4 seconds. The last decimals are stored in the blockette 1001 (see SEED manual, p. 124). The miniseed library [uses this blockette 1001](https://github.com/SeisComP3/seiscomp3/blob/master/src/system/libs/3rd-party/libmseed/msrutils.c#L543) but this blockette isn't currently available in the `isEnd` method. Do you think 1e-4 seconds is good enough or do we have to also parse the blockette 1001?

### CODE REVIEW

**Comment by gempa-jabe** [comment]
That looks good. We will test it, thanks again and have a nice weekend.

**Comment by gempa-jabe** [comment]
I have reproduced the issue and can confirm your fix for the data provided. I will test with other data as well.

**Comment by gempa-jabe** [comment]
I would fix some minor formatting issues and prefix the commit message with ```[trunk]```. Would you want to do that or is it OK when I do it?

File: src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp [hunk 1]
```diff
@@ -431,12 +430,28 @@ bool SDSArchive::isEnd() {
...
 
-	rectime = Time::FromYearDay(fsdh->start_time.year,fsdh->start_time.day);
-	rectime.get(&year,&month,&day);
-	rectime.set(year,month,day,(int)fsdh->start_time.hour,(int)fsdh->start_time.min,
-	            (int)fsdh->start_time.sec,(int)fsdh->start_time.fract);
+	hptime_t starttime = ms_btime2hptime(&fsdh->start_time);
+
+	/* Check if a correction is included and if it has been applied,
+	   bit 1 of activity flags indicates if it has been applied */
+	if (fsdh->time_correct != 0 && !(fsdh->act_flags & 0x02)) {
```
Comment (gempa-jabe) [inline src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp]: Add space after and before parenthesis:
```c++
if ( fsdh->time_correct != 0 && !(fsdh->act_flags & 0x02) )
```

File: src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp [hunk 2]
```diff
@@ -431,12 +430,28 @@ bool SDSArchive::isEnd() {
...
+		starttime += (hptime_t) fsdh->time_correct * (HPTMODULUS / 10000);
+	}
+
+	/* Reduce to Unix/POSIX epoch time and fractional seconds */
+	int64_t isec = MS_HPTIME2EPOCH(starttime);
+	int ifract = (int) (starttime - (isec * HPTMODULUS));
+
+	/* Adjust for negative epoch times */
+	if (starttime < 0 && ifract != 0) {
```
Comment (gempa-jabe) [inline src/trunk/libs/seiscomp3/io/recordstream/sdsarchive.cpp]: Formatting: see above

**Comment by Brtle** [comment]
I fixed the commit message and the formatting issues. If you want to make other changes, you can do it! ;-)

**Comment by gempa-jabe** [comment]
Perfect, thank you!

**Pull request merged**